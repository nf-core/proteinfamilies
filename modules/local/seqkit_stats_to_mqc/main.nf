process SEQKIT_STATS_TO_MQC {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ubuntu:20.04' :
        'nf-core/ubuntu:20.04' }"

    input:
    tuple val(meta), path(stats)

    output:
    tuple val(meta), path("${prefix}_stats_mqc.csv"), emit: mqc
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    cat <<EOF > ${prefix}_stats_mqc.csv
    # id: "seqkit_stats_summary"
    # section_name: "SeqKit Statistics Summary"
    # description: "Statistics about the input protein FASTA file from SeqKit."
    # format: "csv"
    # plot_type: "table"
    File,File Format,Sequence Type,Number of Sequences,Total Length (aa),Minimum Length,Average Length,Maximum Length,Length Q1 (25th percentile),Length Q2 (Median),Length Q3 (75th percentile),Total Gaps,N50 Length,N50 Sequence Count
    EOF

    # Skip header line and extract relevant columns (1â€“14) from tab-separated input
    tail -n +2 ${stats} | cut -f1-14 --output-delimiter=, >> ${prefix}_stats_mqc.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sed: \$(sed --version 2>&1 | sed -n 1p | sed 's/sed (GNU sed) //')
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_stats_mqc.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sed: \$(sed --version 2>&1 | sed -n 1p | sed 's/sed (GNU sed) //')
    END_VERSIONS
    """
}
