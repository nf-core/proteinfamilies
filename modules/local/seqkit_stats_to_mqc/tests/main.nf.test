nextflow_process {

    name "Test Process SEQKIT_STATS_TO_MQC"
    script "../main.nf"
    process "SEQKIT_STATS_TO_MQC"

    tag "seqkit"
    tag "seqkit/stats"

    setup {
        run("SEQKIT_STATS") {
            script "../../../nf-core/seqkit/stats/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id: 'fasta' ],
                    file(params.pipelines_testdata_base_path + 'test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                """
            }
        }
    }

    test("proteinfamilies - fasta") {

        when {
            process {
                """
                input[0] = SEQKIT_STATS.out.stats
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.mqc,
                    process.out.versions,
                    path(process.out.versions[0]).yaml
                    ).match()}
            )
        }
    }

    test("proteinfamilies - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = SEQKIT_STATS.out.stats
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.mqc,
                    process.out.versions,
                    path(process.out.versions[0]).yaml
                ).match()
                }
            )
        }

    }
}
