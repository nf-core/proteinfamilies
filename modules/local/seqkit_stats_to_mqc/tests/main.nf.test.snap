{
    "proteinfamilies - fasta - stub": {
        "content": [
            [
                [
                    {
                        "id": "fasta"
                    },
                    "fasta_stats_mqc.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,791b25a2fb44cdeac69275dd1a9771db"
            ],
            {
                "SEQKIT_STATS_TO_MQC": {
                    "sed": 4.7
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-08-01T15:18:46.42441909"
    },
    "proteinfamilies - fasta": {
        "content": [
            [
                [
                    {
                        "id": "fasta"
                    },
                    "fasta_stats_mqc.csv:md5,5f33427a3dc0b0e0b38df473612e269e"
                ]
            ],
            [
                "versions.yml:md5,791b25a2fb44cdeac69275dd1a9771db"
            ],
            {
                "SEQKIT_STATS_TO_MQC": {
                    "sed": 4.7
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-08-01T15:18:39.362154509"
    }
}